### PR TITLE
Check not only the notAfter field, but also the notBefore field of certs and CRLs.

### PR DESCRIPTION
**Pull request by DDvO**
This is the first chunk extracted from #170.
Extends `OSSL_CMP_expired()` to `OSSL_CMP_cmp_timeframe()`, checking not only the `notAfter` field, but also the `notBefore` field of certs and CRLs.

##### Checklist

- [x] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

**Comment by DDvO** [comment]
@Akretsch, can you please take over reviewing this, trying out if it compiles and merging it if all good.

File: apps/cmp.c [hunk 1]
```diff
@@ -1024,11 +1024,15 @@ static int load_certs_also_pkcs12(const char *file, STACK_OF(X509) **certs,
...
+        cmp = OSSL_CMP_cmp_timeframe(X509_get0_notBefore(cert),
+                                     X509_get0_notAfter (cert), vpm);
+        if (cmp != 0) {
             char *s = X509_NAME_oneline(X509_get_subject_name(cert), NULL, 0);
             BIO_printf(bio_err,
-                     "warning: certificate with subject '%s' has expired\n", s);
+                       "warning: certificate with subject '%s' %s\n", s,
+                       cmp > 0 ? "has expired" : "not yet valid");
             OPENSSL_free(s);
 #if 0
```
Comment (Akretsch) [inline apps/cmp.c]: #if 0 cleanup?

File: crypto/cmp/cmp_vfy.c [hunk 2]
```diff
@@ -326,19 +326,23 @@ int OSSL_CMP_print_cert_verify_cb(int ok, X509_STORE_CTX *ctx)
...
     if ((flags & X509_V_FLAG_USE_CHECK_TIME) != 0) {
         check_time = X509_VERIFY_PARAM_get_time(vpm);
         ptime = &check_time;
+    } else if ((flags & X509_V_FLAG_NO_CHECK_TIME) != 0) {
+        return 0; /* ok */
     }
-
-    return (!(flags & X509_V_FLAG_NO_CHECK_TIME) &&
-            endtime && X509_cmp_time(endtime, ptime) < 0);
+    return (end   != NULL && X509_cmp_time(end  , ptime) < 0) ? +1 :
```
Comment (Akretsch) [inline crypto/cmp/cmp_vfy.c]: drop extra space after "end"

Comment (Akretsch) [inline crypto/cmp/cmp_vfy.c]: nested "?" operators might be confusing, consider 

if (end   != NULL && X509_cmp_time(end  , ptime) < 0) {
 return +1;
} else if (start != NULL && X509_cmp_time(start, ptime) > 0) {
return -1;
} else {
 return 0;
}

File: crypto/cmp/cmp_vfy.c [hunk 3]
```diff
@@ -394,21 +398,25 @@ static int check_kid(X509 *cert, const ASN1_OCTET_STRING *skid, int fn)
...
  */
 static int cert_acceptable(X509 *cert, const OSSL_CMP_MSG *msg,
                            const X509_STORE *ts) {
     X509_NAME *sender_name = NULL;
     X509_VERIFY_PARAM *vpm = NULL;
+    int time_cmp;
 
     vpm = ts != NULL ? X509_STORE_get0_param((X509_STORE *)ts) : NULL;
-    if (cert == NULL || msg == NULL || ts == NULL || vpm == NULL)
+    if (cert == NULL || msg == NULL || (ts != NULL && vpm == NULL))
```
Comment (Akretsch) [inline crypto/cmp/cmp_vfy.c]: why ts must be checked here? Checking vpm should be sufficiant?

File: apps/cmp.c (hunk 1 above)
Comment (DDvO) [reply apps/cmp.c]: Well, cleaning up `#if 0` (and `#if 1`) is unrelated to this PR, but nevertheless good to do.
So I've just done this for all of `cmp.c`.

File: crypto/cmp/cmp_vfy.c (hunk 2 above)
Comment (DDvO) [reply crypto/cmp/cmp_vfy.c]: done

Comment (DDvO) [reply crypto/cmp/cmp_vfy.c]: Good point, done

File: crypto/cmp/cmp_vfy.c (hunk 3 above)
Comment (DDvO) [reply crypto/cmp/cmp_vfy.c]: It would not be sufficient because `vpm == NULL` can be accepted if `ts == NULL`.

**Pull request merged**